web-app/CKS.Web/Controllers/AutocompleteController.cs

```diff
@@ -0,0 +1,112 @@
+using CKS.Web.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using NICE.Search.Common.Interfaces;
+using NICE.Search.Common.Models;
+using NICE.Search.Common.Urls;
+using System;
+using System.Linq;
+using System.Text.Json;
+
+namespace CKS.Web.Controllers
+{
+	[ApiController]
+	[Route("api/[controller]")]
+	public class AutocompleteController : ControllerBase
+	{
+		private const string TitleTypeAheadType = "title";
+
+		private readonly ILogger<AutocompleteController> _logger;
+		private readonly ISearchProvider _provider;
+
+		public AutocompleteController(ILogger<AutocompleteController> logger, ISearchProvider provider)
+		{
```

Comment:
Do we have a CORS policy to ensure no one else can use our endpoint for suggestions? Also have XSS payloads been tested as part of the query parameters to see if there's a reflected attack that's possible?

Reply:
As for CORS, we don't have any `Access-Control-Allow-Origin` header set, so it defaults to the browser default, which is to deny, so we shouldn't need to set anything specific here.

As for XSS, I've tried various things and the autocomplete endpoint is definitely OK because we never return the original query, only things directly from elastic. And for the opensearch endpoint, we do return the original query as the first item in the json, but it uses the build in JSON serialization in .net core which uses encodes properly e.g. you get  `["\u0022\u003Ctest\u003E\u0022"` in the resulting json

Reply:
Beautiful, love it!